test/extended/storage/inline.go

```diff
@@ -0,0 +1,315 @@
+package storage
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/google/uuid"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	k8simage "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+
+	configv1 "github.com/openshift/api/config/v1"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	csiInlineVolProfileLabel = "security.openshift.io/csi-ephemeral-volume-profile"
+	csiSharedResourceDriver  = "csi.sharedresource.openshift.io"
+	podSecurityEnforceError  = "has a pod security enforce level that is lower than"
+)
+
+// This is [Serial] because it modifies a CSIDriver object that is used by multiple tests.
+var _ = g.Describe("[sig-storage][Feature:CSIInlineVolumeAdmission][Serial]", func() {
+	defer g.GinkgoRecover()
+	var (
+		ctx                  = context.Background()
+		baseDir              = exutil.FixturePath("testdata", "storage", "inline")
+		secret               = filepath.Join(baseDir, "secret.yaml")
+		csiSharedSecret      = filepath.Join(baseDir, "csi-sharedsecret.yaml")
+		csiSharedRole        = filepath.Join(baseDir, "csi-sharedresourcerole.yaml")
+		csiSharedRoleBinding = filepath.Join(baseDir, "csi-sharedresourcerolebinding.yaml")
+		csiInlineSCCPatch    = filepath.Join(baseDir, "csi-inline-scc-patch.yaml")
+
+		beforeEach = func(oc *exutil.CLI) {
+			if !isTechPreviewNoUpgrade(oc) {
+				g.Skip("this test is only expected to work with TechPreviewNoUpgrade clusters")
+			} else {
+				// TODO: remove this when CSI volumes are allowed by the default SCC's (i.e. after TechPreview)
+				g.By("adding restricted-v2 SCC permission to use inline CSI volumes")
+				err := oc.AsAdmin().Run("patch").Args("scc/restricted-v2", "--type", "merge", "--patch-file", csiInlineSCCPatch).Execute()
```

Comment:
Is this reverted after the tests?

Reply:
Good question, I added a new commit to generate a dynamic patch and revert it at the end of the test, so it only changes if 'csi' is missing from the list. This handles other changes to the default SCC much better than the static yaml approach.